packages/signing-key/src.ts/index.ts

```diff
@@ -0,0 +1,169 @@
+"use strict";
+
+import { EC, ED } from "./elliptic";
+
+import { arrayify, BytesLike, hexlify, hexZeroPad, Signature, SignatureLike, splitSignature } from "@ethersproject/bytes";
+import { defineReadOnly } from "@ethersproject/properties";
+
+import { Logger } from "@hethers/logger";
+import { version } from "./_version";
+const logger = new Logger(version);
+
+let _ecCurve: EC = null
+let _edCurve: ED = null
+function getECCurve() {
+    if (!_ecCurve) {
+        _ecCurve = new EC("secp256k1");
+    }
+    return _ecCurve;
+}
+function getEDCurve() {
+    if (!_edCurve) {
+        _edCurve = new ED("ed25519");
+    }
+    return _edCurve;
+}
+
+
+
+export class SigningKeyED {
```

Comment:
Is it a good idea to rename the keys based on their types eg SigningKeyED25519 and SigningKeyECDSA?

Reply:
Yes, It is a good idea but if we do it, we'll need to increase the version of the library because Singing Key is exported globally via hethers.utils